servicetalk-transport-netty/src/main/java/io/servicetalk/transport/netty/NettyIoExecutors.java

```diff
@@ -55,10 +72,25 @@ public static IoExecutor createIoExecutor(int ioThreads) {
     /**
      * Creates a new {@link IoExecutor} with the default number of {@code ioThreads}.
      *
-     * @param threadFactory the {@link ThreadFactory} to use. If possible you should use an instance
-     * of {@link IoThreadFactory} as it allows internal optimizations.
+     * @param <T> Type of the IO thread instances created by factory.
+     * @param threadFactory the {@link io.servicetalk.transport.api.IoThreadFactory} to use.
+     * @return The created {@link IoExecutor}
+     */
+    public static <T extends Thread & IoThread> IoExecutor createIoExecutor(
+            io.servicetalk.transport.api.IoThreadFactory threadFactory) {
+        return io.servicetalk.transport.netty.internal.NettyIoExecutors.createIoExecutor(threadFactory);
```

Comment:
I've prepared a patch instead of comments:

```patch
Index: servicetalk-transport-netty/src/main/java/io/servicetalk/transport/netty/NettyIoExecutors.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/servicetalk-transport-netty/src/main/java/io/servicetalk/transport/netty/NettyIoExecutors.java b/servicetalk-transport-netty/src/main/java/io/servicetalk/transport/netty/NettyIoExecutors.java
--- a/servicetalk-transport-netty/src/main/java/io/servicetalk/transport/netty/NettyIoExecutors.java	(revision c8bad3aac7ec362fb97e07e84e1f0e8f2633162b)
+++ b/servicetalk-transport-netty/src/main/java/io/servicetalk/transport/netty/NettyIoExecutors.java	(date 1629224416098)
@@ -16,8 +16,8 @@
 package io.servicetalk.transport.netty;
 
 import io.servicetalk.transport.api.IoExecutor;
+import io.servicetalk.transport.api.IoThreadFactory;
 import io.servicetalk.transport.api.IoThreadFactory.IoThread;
-import io.servicetalk.transport.netty.internal.IoThreadFactory;
 import io.servicetalk.transport.netty.internal.NettyIoExecutor;
 
 import java.util.concurrent.ThreadFactory;
@@ -37,9 +37,7 @@
      * @param ioThreads number of threads.
      * @param threadFactory the {@link ThreadFactory} to use.
      * @return The created {@link IoExecutor}
-     * @deprecated Future versions of ServiceTalk will require a {@link io.servicetalk.transport.api.IoThreadFactory}
-     * for creating {@link IoExecutor} threads, use
-     * {@link #createIoExecutor(int, io.servicetalk.transport.api.IoThreadFactory)} instead.
+     * @deprecated Use {@link #createIoExecutor(int, IoThreadFactory)}.
      */
     @Deprecated
     public static IoExecutor createIoExecutor(int ioThreads, ThreadFactory threadFactory) {
@@ -51,11 +49,11 @@
      *
      * @param <T> Type of the IO thread instances created by factory.
      * @param ioThreads number of threads.
-     * @param threadFactory the {@link io.servicetalk.transport.api.IoThreadFactory} to use.
+     * @param threadFactory the {@link IoThreadFactory} to use.
      * @return The created {@link IoExecutor}
      */
     public static <T extends Thread & IoThread> IoExecutor createIoExecutor(int ioThreads,
-            io.servicetalk.transport.api.IoThreadFactory threadFactory) {
+            IoThreadFactory<T> threadFactory) {
         return io.servicetalk.transport.netty.internal.NettyIoExecutors.createIoExecutor(ioThreads, threadFactory);
     }
 
@@ -66,18 +64,19 @@
      * @return The created {@link IoExecutor}
      */
     public static IoExecutor createIoExecutor(int ioThreads) {
-        return createIoExecutor(ioThreads, newIoThreadFactory());
+        return createIoExecutor(ioThreads,
+                // FIXME: change to NettyIoThreadFactory when it's available
+                new io.servicetalk.transport.netty.internal.IoThreadFactory(NettyIoExecutor.class.getSimpleName()));
     }
 
     /**
      * Creates a new {@link IoExecutor} with the default number of {@code ioThreads}.
      *
      * @param <T> Type of the IO thread instances created by factory.
-     * @param threadFactory the {@link io.servicetalk.transport.api.IoThreadFactory} to use.
+     * @param threadFactory the {@link IoThreadFactory} to use.
      * @return The created {@link IoExecutor}
      */
-    public static <T extends Thread & IoThread> IoExecutor createIoExecutor(
-            io.servicetalk.transport.api.IoThreadFactory threadFactory) {
+    public static <T extends Thread & IoThread> IoExecutor createIoExecutor(IoThreadFactory<T> threadFactory) {
         return io.servicetalk.transport.netty.internal.NettyIoExecutors.createIoExecutor(threadFactory);
     }
 
@@ -86,9 +85,7 @@
      *
      * @param threadFactory the {@link ThreadFactory} to use.
      * @return The created {@link IoExecutor}
-     * @deprecated Future versions of ServiceTalk will require a {@link io.servicetalk.transport.api.IoThreadFactory}
-     * for creating {@link IoExecutor} threads, use
-     * {@link #createIoExecutor(io.servicetalk.transport.api.IoThreadFactory)} instead.
+     * @deprecated Use {@link #createIoExecutor(IoThreadFactory)}.
      */
     @Deprecated
     public static IoExecutor createIoExecutor(ThreadFactory threadFactory) {
@@ -101,10 +98,8 @@
      * @return The created {@link IoExecutor}
      */
     public static IoExecutor createIoExecutor() {
-        return createIoExecutor(newIoThreadFactory());
-    }
-
-    private static IoThreadFactory newIoThreadFactory() {
-        return new IoThreadFactory(NettyIoExecutor.class.getSimpleName());
+        return createIoExecutor(
+                // FIXME: change to NettyIoThreadFactory when it's available
+                new io.servicetalk.transport.netty.internal.IoThreadFactory(NettyIoExecutor.class.getSimpleName()));
     }
 }

```